packages/interface-adapter/lib/interface-adapter.ts

```diff
@@ -46,7 +48,11 @@ export class InterfaceAdapter {
     return this.adapter.getBlock(block);
   }
 
-  public setProvider(provider: Provider) {
+  public setProvider(provider: Provider): Promise<void> {
```

Comment:
is `adapter.setProvider` not being used? we'll need to add `await` to those (this is the part that I figured you might object to :)

Reply:
we only use it in `@truffle/contract`'s `setProvider` method. I believe at least one of the CI jobs would be failing if it wasn't properly resolving.

Reply:
anddd it's gone.